src/pretty-dom.js

```diff
@@ -25,6 +25,11 @@ function prettyDOM(
   maxLength = getMaxLength(dom),
   options,
 ) {
+  if (!(dom instanceof HTMLDocument) && !(dom instanceof HTMLElement)) {
```

Comment:
I suppose the error is thrown at `dom.outerHTML.length` which means `Element` is sufficient. Otherwise we would also throw on svg elements.

Do we need to consider cross-realm instanceof checks here as well? Getting the window of an element would not work if we didn't get an Element in the first place. I think checking for `typeof dom.outerHTML === 'string'` might be sufficient here in asserting that the given value implements `Element`.

Reply:
> I suppose the error is thrown at `dom.outerHTML.length` which means `Element` is sufficient. Otherwise we would also throw on svg elements.

Unfortunately it isn't because `document` isn't an instance of `Element`

> I think checking for `typeof dom.outerHTML === 'string'` might be sufficient here in asserting that the given value implements `Element`.

I absolutely agree that `typeof dom.outerHTML === 'string'` checking is enough however it may hide a real problem from a developer

Reply:
> Unfortunately it isn't because document isn't an instance of Element

Which is why you would keep the HTMLDocument check.

Cross-realm problem still exists.

> I absolutely agree that typeof dom.outerHTML === 'string' checking is enough however it may hide a real problem from a developer

That would be? 

IMO runtime type checking is not necessary here. If people care about type checking then you already have very good solutions with flow or typescript.

Reply:
Types may lies. Right now `dom` typed as `HTMLElement` 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/403ac51161e5773d0ec7eb965004e18e9d4a03c8/types/testing-library__dom/pretty-dom.d.ts#L3

However there is a test for rendering `document`

https://github.com/testing-library/dom-testing-library/blob/a401bb50c8301c27b2698d299b3ebc1483cc7633/src/__tests__/pretty-dom.js#L39-L40

Runtime checking isn't a silver bullet however it allows to fail fast when a library's user use it as not expected and teaches she how to do it right.

Reply:
> Types may lies

Did it occur to you that it was a mistake instead of an intention to specify something wrong?

Reply:
Sure, it did.

For me as a developer it's so cool to have strict contracts in a library. What do you think about an alternative conditional here like:

```diff
- if (!(dom instanceof HTMLDocument) && !(dom instanceof HTMLElement)) {
-     throw new TypeError(
-      'Debuggable value should be an instance of HTMLDocument or HTMLElement.',
-    )
- }
+ if (!('outerHTML' in dom)) {
+     throw new TypeError(
+      'Debuggable object should have `outerHTML` field like an HTMLElement instance.',
+    )
+ }

```

Reply:
Sounds better but you're still using `HTMLElement` which is too narrow. No need throw so many technicallaties around "Expected an element or document but got X".

Reply:
Thanks for suggestion. I've updated this check and add a test